rails/app/policies/api/v1/external_activity_policy.rb

```diff
@@ -12,4 +12,8 @@ def create?
     user && (user.is_project_admin? || user.is_project_researcher? || user.has_role?('manager','admin','researcher'))
   end
 
+  def update_by_url?
+    user && (user.is_project_admin? || user.is_project_researcher? || user.has_role?('manager','admin','researcher','author'))
+  end
+
```

Comment:
This is too permissive. It would allow any author to update any external activity.
It should only allow authors to update their own external activities. 
I'm actually not sure this check is used. I think that the `authorize external_activity` in `rails/app/controllers/api/v1/external_activities_controller.rb` will not use this policy instead it will use `rails/app/policies/external_activity_policy.rb`. 

So I'd try deleting this new policy method and see if things still work.